fix: misc setup_wizard region slide

saxenabhishek opened the pull request:
# Before 

<div align="center">
    <img src="https://user-images.githubusercontent.com/33656173/146039388-003108ba-adbe-48dd-ac14-b5d3cd6d49b0.png" height = "500">

<p>
    <i>
         Long list of countries, default text is empty, no placeholder
    </i>
</p>

</div>




# Changes

- placeholder on all fields of region slides
- autocomplete on the country input field
- default selected values on inputs instead of blank values


# After
<div align="center">
    <div align="center">
        <img src="https://user-images.githubusercontent.com/33656173/146396734-849f7a84-bd86-4242-ab26-977df764358c.png" height = "350"/>
        <img src="https://user-images.githubusercontent.com/33656173/146039455-5ec19e04-51a3-42d5-8486-feb74ded0a73.jpg" height = "350"/>
 <img src="https://user-images.githubusercontent.com/33656173/146396747-d7a19be3-e37c-4f0b-a824-a252c3a9082d.png" height = "350"/>
    </div>
    <p>
        <i>
            `select county` is the placeholder, country field now has autocomplete
        </i>
    </p>
</div>

surajshetty3416 commented on frappe/desk/page/setup_wizard/setup_wizard.js:
```diff
@@ -529,16 +533,19 @@ frappe.setup.utils = {
+			.empty()
+			.add_options(
+				frappe.utils.unique(
+					["Select Currency"].concat($.map(data.country_info, opts => opts.currency).sort())
```
Why is "Select Country/Timezone/Currency" set as an option if we are adding a placeholder?

saxenabhishek replied:
Hmm, they were default options before, but I think the county field doesn't need one now because it's a autocomplete field. A placeholder will be enough. 
TImezone and currency fields have default selected options as soon as  on_load() runs. But the placeholder is on-screen till then.

saxenabhishek committed:
fix: misc setup_wizard region slide

placeholder on all fields
autocomplete on country input
default selected values on inputs

surajshetty3416 committed:
refactor: Use placeholder instead of additional option in select

- Additional options were still selectable which could have got through
- Minor formatting changes

surajshetty3416 commented:
@saxenabhishek Pushed [a change](https://github.com/frappe/frappe/pull/15277/commits/1dd870ce73457b1505e74c706be14ea8a7a33867) to use a placeholder for select field instead of an additional select option.

<img width="869" alt="Screenshot 2021-12-21 at 8 50 33 AM" src="https://user-images.githubusercontent.com/13928957/146866040-009599dd-443f-4b2e-8e10-87aeb9d65e04.png">

Works as expected.

surajshetty3416 committed:
revert: Empty value checks

ankush commented:
This is breaking setup wizard in v13 😐

ankush commented:
@mergifyio backport version-13-hotfix
